plutus-core/cost-model/test/TestCostModels.hs

```diff
@@ -38,203 +44,255 @@ import           Unsafe.Coerce                                  (unsafeCoerce)
    CostModelCreation to convert R results to picoseconds expressed as
    CostingIntegers.  To deal with (A), we don't check for exact equality of the
    outputs but instead check that the R result and the Haskell result agreee to
-   within a factor of 1/10000 (one hundredth of a percent).
+   within a factor of 1/100 (one percent).
 
-   This executes all of the R code (reading the CSV file and constructing all of
-   the models) for every instance of every test, so it takes a moderately long
-   time (maybe 3 minutes).
 -}
 
+-- How many tests to run for each costing function
+numberOfTests :: TestLimit
+numberOfTests = 100
+
+-- | Generate inputs for costing functions, making sure that we test a large
+-- range of inputs, but that we also get small inputs.
+memUsageGen :: Gen CostingInteger
```

Comment:
I'm beginning to hate Hedgehog generators.  I think the previous version was mostly only generating 0,1, and 2.